src/vm.rs

```diff
@@ -204,6 +207,33 @@ impl<VirtBackend: VirtualizationBackend> UhyveVm<VirtBackend> {
 		#[cfg(not(target_os = "linux"))]
 		let mut mem = MmapMemory::new(0, memory_size, guest_address, false, false);
 
+		// TODO: file_mapping not in kernel_info
+		let file_mapping = Mutex::new(UhyveFileMap::new(&params.file_mapping, &params.tempdir));
+
+		let serial = UhyveSerial::from_params(&params.output)?;
+
+		// Takes place before the kernel is actually loaded.
+		#[cfg(target_os = "linux")]
+		{
+			if params.file_isolation != FileSandboxMode::None {
+				let mut host_paths = file_mapping.lock().unwrap().get_all_host_paths();
+				let temp_dir = file_mapping
+					.lock()
+					.unwrap()
+					.get_temp_dir()
+					.unwrap()
+					.to_owned();
+				let landlock = initialize_landlock_vm(
+					params.file_isolation,
+					kernel_path.to_str().unwrap().to_owned(),
+					&params.output,
+					&mut host_paths,
+					temp_dir,
+				);
+				landlock.apply_landlock_restrictions();
+			}
```

Comment:
Can we move this to _isolation/mod.rs_

Reply:
The idea behind this part of the code is that we "clearly" show to the developer what kinds of paths are passed to the Landlock submodule. I don't think it would be pretty if we were to pass e.g. the `file_mapping` variable and extracted all the paths and the temporary directory, because this spreads a lot of UhyveFileMap-specific logic (which is subject to change) across Uhyve.

Wouldn't it better to break this into a new function within `UhyveVm` instead? That would address both problems at once.

Reply:
See: https://github.com/hermit-os/uhyve/pull/816/commits/7601c4242ffcd9bc752471a6047475b49af990a8